pandas/tests/indexing/test_loc.py

```diff
@@ -969,6 +969,24 @@ def test_loc_reverse_assignment(self):
 
         tm.assert_series_equal(result, expected)
 
+    def test_loc_setting_value_at_multiindex(self):
+        # issue: https://github.com/pandas-dev/pandas/issues/34870
+
+        arrays = [['bar', 'bar', 'baz', 'baz', 'foo', 'foo', 'qux', 'qux'],
+                  ['one', 'two', 'one', 'two', 'one', 'two', 'one', 'two']]
+
+        tuples = list(zip(*arrays))
+        index = pd.MultiIndex.from_tuples(tuples, names=['first', 'second'])
+
+        dt = pd.Series(np.random.randn(8), index=index)
+        correct = dt.copy(deep=True)
+        test = dt.copy(deep=True)
+
+        test.loc[('baz', 'one'):('foo', 'two')] = 4
+        correct[('baz', 'one'):('foo', 'two')] = 4
+
+        assert correct.equals(test)
```

Comment:
Can you explicitly construct the expected output (i.e., rather than setting with another form of indexing) and then use assert_series_equal to test this (naming the two Series result and expected)?